tests/test_examples.py

```diff
@@ -299,12 +307,21 @@ def test(self):
                 example_script = example_script[0]
 
             # The ESMFold example has no arguments, so we can execute it right away
-            if self.EXAMPLE_NAME == "run_esmfold":
-                p = subprocess.Popen(["python3", example_script])
+            if self.EXAMPLE_NAME in ["run_esmfold", "run_zero_shot_eval"]:
+                p = subprocess.Popen(["python3", example_script], stderr=subprocess.PIPE)
                 return_code = p.wait()
 
                 # Ensure the run finished without any issue
                 self.assertEqual(return_code, 0)
+                if self.EXAMPLE_NAME == "run_zero_shot_eval":
+                    out = p.stderr.readlines()
+                    for line in out:
+                        if b"accuracy" in line:
+                            reg = re.compile("Zero-shot accuracy: (?P<accuracy>.+)")
+                            match = reg.search(line.decode("utf-8"))
+                            if match:
+                                accuracy = float(match.groupdict().get("accuracy"))
+                                self.assertTrue(accuracy > 0.43)
```

Comment:
Not a big fan of parsing the output of stderr. Let's rather write the results in a JSON file in the example (you can do something similar to https://github.com/huggingface/optimum-habana/blob/9f6eba319056aec1d9806b9eb8c377b88b7ef3ce/examples/stable-diffusion/training/textual_inversion.py#L965) and parse it in the CI script.